airflow/providers/postgres/hooks/postgres.py

```diff
@@ -320,6 +320,36 @@ def _generate_insert_sql(
 
         return sql
 
+    def ingest_embedding(
+        self, table: str, input_data: list[str], embeddings: list[float], vector_size: int
```

Comment:
Instead of accepting two lists I feel it’d be more natural to accept a list of two tuples. Or better, an iterable since that’s the only required interface (for the for loop).

i.e.

```suggestion
        self,
        table: str,
        input_data: Iterable[tuple[str, float]],
        vector_size: int,
```